### PR TITLE
build: bump @flybywiresim/eslint-config to 0.2.2

### PR DESCRIPTION
**Pull request by Benjozork**
## Summary of Changes

* Waiting for #5541
* Bump `@flybywiresim/eslint-config` to 0.2.2.

Discord username (if different from GitHub): someperson#4953

## Testing Instructions

- Make sure **all** flyPad features still work
- Make sure DCDU still has same very basic functionality
- Make sure SD pages still work

## How to download the PR for QA

Every new commit to this PR will cause a new A32NX artifact to be created, built, and uploaded.

1. Make sure you are signed in to GitHub
1. Click on the **Checks** tab on the PR
1. On the left side, click on the bottom **PR** tab
1. Click on the **A32NX** download link at the bottom of the page

### CODE REVIEW

**Comment by Benjozork** [comment]
cc @Lucky38i

**Comment by davidwalschots** [comment]
Created 19 days ago in a failed lint state. What are we doing with this PR?

**Comment by Benjozork** [comment]
@davidwalschots I meant to make this a draft, I am waiting on #5541 since Erick wanted to do the refactoring part and in the mean time conform to the new ESLint rules.

**Review by bouveng — CHANGES_REQUESTED** [review]
Hi! Is this PR supposed to contain work on DCDU, PFD, RMP, EICAS and EFB?

**Comment by Benjozork** [comment]
Yes, it fixes lint errors and warnings.

**Comment by Benjozork** [comment]
Converting to draft since not done with warnings yet

**Comment by Benjozork** [comment]
Only few errors/warnings to go...

**Comment by Benjozork** [comment]
DONE!!!!!

**Comment by tracernz** [comment]
> 
> 
> DONE!!!!!

CI doesn't agree. RIP. 🥲

File: package.json [hunk 1]
```diff
@@ -23,7 +23,7 @@
     "@babel/preset-env": "^7.12.1",
     "@babel/preset-react": "^7.12.1",
     "@babel/preset-typescript": "~7.12.7",
-    "@flybywiresim/eslint-config": "~0.1.0",
+    "@flybywiresim/eslint-config": "^0.2.3",
```
Comment (beheh) [inline package.json:26]: I just realized we don't have a lock file 🤔

In any case, I would suggest sticking with the `~`-style dependency, then it wil only upgrade to 0.2.4, 0.2.5... but not to 0.3.0. That seems safer.

File: src/instruments/src/EFB/Dashboard/Dashboard.tsx [hunk 2]
```diff
@@ -52,7 +52,8 @@ const Dashboard: React.FC<DashboardProps> = (props) => {
                     </div>
 
                     <div className="h-3/5 ml-3 mt-3 rounded-2xl shadow-lg">
-                        <Map currentFlight={getCurrentFlightData} forceTileset="carto-dark" disableMenu hideOthers />
+                        {/* eslint-disable-next-line react/jsx-handler-names */}
```
Comment (beheh) [inline src/instruments/src/EFB/Dashboard/Dashboard.tsx:55]: I think either we should enforce this rule and apply it everywhere, or we should remove the rule from the config. Or is there anything special about the usage here that warrants an exception?

File: src/instruments/src/EFB/TODCalculator/GroundSpeed/GroundSpeedAuto/GroundSpeedAuto.tsx [hunk 3]
```diff
@@ -16,21 +16,17 @@ const GroundSpeedAuto = ({ groundSpeedData, currentAltitude, setTodGroundSpeed,
     let [simGroundSpeed] = useSimVar('GPS GROUND SPEED', 'knots', 1_000);
     simGroundSpeed = round(simGroundSpeed);
 
-    const setCurrentGroundSpeed = () => {
+    const { groundSpeed } = last(groundSpeedData);
```
Comment (beheh) [inline src/instruments/src/EFB/TODCalculator/GroundSpeed/GroundSpeedAuto/GroundSpeedAuto.tsx:19]: While this PR doesn't really change this, this behaviour seems dodgy for two reasons:
- If you're storing data in a React component, that is state, and should be treated as such (useState over last)
- Storing past prop values can be dodgy for the same reasons as prevTime in the clock components above. I'm confused why this is even necessary in this component.

File: src/instruments/src/RMP/Components/StandbyFrequency.tsx [hunk 4]
```diff
@@ -99,22 +99,23 @@ const offsetFrequencyChannel = (spacing: ChannelSpacing, channel: number, offset
  * Hooks to outer and inner rotary encoder knobs.
  * Renders standby frequency RadioPanelDisplay sub-component.
  */
-export const StandbyFrequency = (props: Props) => {
+export const StandbyFrequency:React.FC<Props> = ({ side, value, setValue }) => {
```
Comment (beheh) [inline src/instruments/src/RMP/Components/StandbyFrequency.tsx:102]:
```suggestion
export const StandbyFrequency: React.FC<Props> = ({ side, value, setValue }) => {
```

**Review by beheh — CHANGES_REQUESTED** [review]
Mostly typos, but a few . Some of the hook dependency changes could definitely uncover or change component behaviours, especially in the Clock and EFB from what I could see.

One more tip: The changes are correct in that technically if you call a setter e.g. `setMyStateVariabe` in useEffect, its a dependency. However, it's a (not very well documented) React convention that setter coming from another hook should not change and are thus usually excluded from dependency arrays (unless e.g. you pass a prop called setX, that might have to be passed in!). Usually I would expect eslint to pick that up itself and to not add variables starting with `set` coming from a hook to the dependency array. However, I'm seeing that it's inconsistentally applied in this PR - sometimes they're there, sometimes not.

File: src/instruments/src/EFB/Dashboard/Dashboard.tsx (hunk 2 above)
Comment (Benjozork) [reply src/instruments/src/EFB/Dashboard/Dashboard.tsx:55]: This is because:

1. a aprop with prefix `on` has to take a function prefixed with `on|handle`;
2. a prop being passed a function prefixed with `on|handle` must be prefixed with `on`.

The mistake here was re-naming `getCurrentFlightData` to `handleGettingCurrentFlightData,` which is incorrect since it is not an event handler.

I will fix this.

File: package.json (hunk 1 above)
Comment (Benjozork) [reply package.json:26]: From npm docs:

image

TL;DR: `^` does not allow minor version updates when major version is 0.

File: src/instruments/src/Common/hooks.tsx [hunk 5]
```diff
@@ -57,7 +57,7 @@ export const useInteractionEvents = (events: string[], handler: (any?) => void):
         return () => {
             events.forEach((event) => getRootElement().removeEventListener(event, wrappedHandler));
         };
-    }, [
-        ...events,
+    }, [ /* eslint-disable-line react-hooks/exhaustive-deps */
+        ...events, /* eslint-disable-line react-hooks/exhaustive-deps */
     ]);
```
[the remark this answers is not in the document]
Comment (Benjozork) [reply src/instruments/src/Common/hooks.tsx:62]: This is caused by ESLint not being able to verify the dependency array, since we are using a spread operator.

image

I can't really think of a good way to fix this, other than create a custom hook to memoize the dependencies that would have an ESLint exception inside it to circumvent the warning in the call site.

File: src/instruments/src/Common/simVars.tsx [hunk 6]
```diff
@@ -296,8 +296,10 @@ export const useSimVar = (
         return [0, () => {}];
     }
 
+    /* eslint-disable react-hooks/rules-of-hooks */
     const value = useSimVarValue(name, unit, maxStaleness);
     const setter = useSimVarSetter(name, unit);
+    /* eslint-enable react-hooks/rules-of-hooks */
```
[the remark this answers is not in the document]
Comment (Benjozork) [reply src/instruments/src/Common/simVars.tsx]: Removed `SIMVAR_DISABLE`. ACE does not use it.

File: src/instruments/src/Common/simVars.tsx [hunk 7]
```diff
@@ -486,8 +488,10 @@ export const useSplitSimVar = (
         return [0, () => {}];
     }
 
+    /* eslint-disable react-hooks/rules-of-hooks */
     const value = useSimVarValue(readName, readUnit, maxStaleness);
     const setter = useSimVarSetter(readName, writeUnit || readUnit, writeName);
+    /* eslint-enable react-hooks/rules-of-hooks */
```
[the remark this answers is not in the document]
Comment (Benjozork) [reply src/instruments/src/Common/simVars.tsx]: See above.

File: src/instruments/src/EFB/ATC/ATC.tsx [hunk 8]
```diff
@@ -77,9 +72,13 @@ export const ATC = () => {
         loadAtc();
     }, [loadAtc]);
 
+    /* eslint-disable react-hooks/exhaustive-deps */
```
[the remark this answers is not in the document]
Comment (Benjozork) [reply src/instruments/src/EFB/ATC/ATC.tsx]: Fixed.

File: src/instruments/src/RMP/Components/StandbyFrequency.tsx (hunk 4 above)
Comment (Benjozork) [reply src/instruments/src/RMP/Components/StandbyFrequency.tsx]: Fixed.

File: src/instruments/src/Common/hooks.tsx (hunk 5 above)
[the remark this answers is not in the document]
Comment (beheh) [reply src/instruments/src/Common/hooks.tsx:62]: To be specific I really meant the double comment. The spread operator seems acceptable to me.

[the remark this answers is not in the document]
Comment (Benjozork) [reply src/instruments/src/Common/hooks.tsx]: Fixed

**Review by beheh — APPROVED** [review]
All the critical feedback has been addressed and this would seem ready for testing.

**Comment by xZreai** [comment]
@Benjozork Any specific testing instructions?

**Comment by Benjozork** [comment]
Added.

**Comment by ShadowFrost3** [comment]
I had (haven't retested since then) some serious FPS issues with this build. (Asked in discord) Any word on whether this issue is limited to me?

**Comment by Benjozork** [comment]
No, I had the issue too, but I cannot find the cause. Waiting on help from some other people.

**Comment by Benjozork** [comment]
EFB infinite rendering issues, will wait until flypadosv3 because I was not able to figure them out for now.

**Comment by Benjozork** [comment]
Replaced by #7344
